### PR TITLE
Add message log

### PR DESCRIPTION
**Pull request by windymilla**
Outputs messages with a timestamp to a dialog as well as to stderr. Shows the dialog if currently hidden.

Currently supports errors. Other message types could be added later if needed.

### CODE REVIEW

**Comment by windymilla** [comment]
Testing notes:
An easy way to cause an error is to edit the `.json`  file that goes with a project file (formerly "bin" file), e.g. adding/removing a comma.

File: src/guiguts/mainwindow.py [hunk 1]
```diff
@@ -593,6 +582,99 @@ def add_binding(self, key: str, event: str, callback: Callable[[], None]) -> Non
...
+
+        self.messagelog = ScrolledReadOnlyText(frame, wrap=tk.NONE)
+        self.messagelog.grid(column=0, row=0, sticky="NSEW")
+
+    def append(self, message: str) -> None:
+        """Append a message to the message log dialog."""
+        self.messagelog.insert("end", message)
+
+
+class MessageLog:
```
Comment (cpeel) [inline src/guiguts/mainwindow.py:643]: We might want to think about introducing standard python logging in the code and have one of the handlers output errors to the `MessageLogDialog`. That could happen separately after this PR however.

**Review by cpeel — APPROVED** [review]
Looks good to me caveat changing to standard python logging down the road.

**Comment by srjfoo** [comment]
> Testing notes: An easy way to cause an error is to edit the `.json` file that goes with a project file (formerly "bin" file), e.g. adding/removing a comma. Then try opening the project file.

That certainly caused the error window to pop up. I noticed that it did load the CTF, just didn't load the images. I can see why the images don't load, but wanted to make sure that, in this particular case, the text loading is the expected behavior.

**Comment by windymilla** [comment]
I think that's ok, yes. It should only happen if the user messes with their json file. Thanks

File: src/guiguts/mainwindow.py (hunk 1 above)
Comment (windymilla) [reply src/guiguts/mainwindow.py:643]: I've added an issue #94 to use python logging - looks like just what we need, thanks.

**Review by srjfoo — APPROVED** [review]

**Pull request merged**